websites/H/HBO Max/presence.ts

```diff
@@ -47,48 +47,85 @@ const presence = new Presence({
 	},
 	coverUrls: Record<string, string> = {};
 
-/* eslint-disable camelcase */
-function getToken(): Promise<string> {
-	return new Promise(resolve => {
-		fetch("https://oauth.api.hbo.com/auth/tokens", {
-			method: "POST",
-			headers: {
-				"Content-Type": "application/json"
-			},
-			body: JSON.stringify({
-				client_id: "585b02c8-dbe1-432f-b1bb-11cf670fbeb0",
-				client_secret: crypto.randomUUID(),
-				scope: "browse video_playback",
-				grant_type: "client_credentials",
-				deviceSerialNumber: crypto.randomUUID(),
-				clientDeviceData: {
-					paymentProviderCode: "blackmarket"
-				}
-			})
+let isFetching = false;
+
+function fetchToken(): Promise<string> {
+	return fetch("https://oauth.api.hbo.com/auth/tokens", {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json"
+		},
+		/* eslint-disable camelcase */
+		body: JSON.stringify({
+			client_id: "585b02c8-dbe1-432f-b1bb-11cf670fbeb0",
+			client_secret: crypto.randomUUID(),
+			scope: "browse video_playback",
+			grant_type: "client_credentials",
+			deviceSerialNumber: crypto.randomUUID(),
+			clientDeviceData: {
+				paymentProviderCode: "blackmarket"
+			}
+		})
+		/* eslint-enable camelcase */
+	})
+		.then(res => res.json())
+		.then(res => res.access_token);
+}
+
+function fetchClientConfig(
+	token: string
+): Promise<{ routeKey: string; countryCode: string }> {
+	return fetch("https://sessions.api.hbo.com/sessions/v1/clientConfig", {
+		method: "POST",
+		headers: {
+			authorization: `Bearer ${token}`,
+			"content-type": "application/json"
+		},
+		body: JSON.stringify({
+			contract: "abc:1.0.0.0",
+			preferredLanguages: ["en-us"]
 		})
-			.then(x => x.json())
-			.then(x => resolve(x.access_token));
-	});
+	})
+		.then(res => res.json())
+		.then(res => {
+			return {
+				routeKey: res.routeKeys.contentSubdomain,
+				countryCode: new URLSearchParams(
+					res.features["express-content"].config.expressContentParams
+				).get("country-code")
+			};
+		});
```

Comment:
Formatting might be a lil off
```suggestion
		.then(res => ({
				routeKey: res.routeKeys.contentSubdomain,
				countryCode: new URLSearchParams(
					res.features["express-content"].config.expressContentParams
				).get("country-code")
			}));
```